cluster/cluster.go

```diff
@@ -122,6 +100,47 @@ func (c *Cluster) NewClient(serviceName string) (*Client, error) {
 	return newClient(c.localAddr, serviceName, c.Registry)
 }
 
+func memberAdd(ctx context.Context, client *clientv3.Client, cfg embed.Config) (string, error) {
+	peerUrlStrings := make([]string, len(cfg.LPUrls))
+	for i, url := range cfg.LPUrls {
+		peerUrlStrings[i] = url.String()
+	}
+
+	mresp, err := client.MemberAdd(ctx, peerUrlStrings)
+	if err != nil {
+		return "", fmt.Errorf("failed to add member with peerURLs %v: %w", cfg.LPUrls, err)
+	}
+
+	initialClusterStrings := make([]string, 0, 1+len(cfg.LPUrls))
+	for _, url := range cfg.LPUrls {
+		initialClusterStrings = append(initialClusterStrings, initialClusterStringFormatter(cfg.Name, url.String()))
+	}
+	for _, member := range mresp.Members {
+		if member.Name != "" {
```

Comment:
since the member's etcd is spun up after the member is added, the name isn't instantiated until later so the previous for loop handles adding the node name based on the config given.